src/plugins/content_management/docs/content_onboarding.md

```diff
@@ -0,0 +1,678 @@
+  {
+    item: mapItemSchema,
+    meta: schema.object(
+      {
+        someOptionalMetaField: schema.maybe(schema.string()), // See "MapGetOut" above for this meta field
+      },
+      { unknowns: 'forbid' }
+    ),
+  },
+  { unknowns: 'forbid' }
+);
+
+// Schema for the "CreateOptions" TS interface
+const createOptionsSchema = schema.object({
+  references: schema.maybe(referencesSchema),
+});
+
+// ... follow the same pattern for all your objects
+
+// Create a CM services definition
+export const serviceDefinition: ServicesDefinition = {
+  get: {
+    out: {
+      result: {
+        schema: getResultSchema,
+      },
+    },
+  },
+  create: {
+    in: {
+      options: {
+        schema: createOptionsSchema,
+      },
+      data: {
+        // Schema to validate the data to be saved
+        schema: mapAttributesSchema,
+      },
+    },
+    out: {
+      result: {
+        schema: schema.object(
+          {
+            item: mapSavedObjectSchema,
+          },
+          { unknowns: 'forbid' }
+        ),
+      },
+    },
+  },
+  // ...other methods
+};
+```
+
+#### 2.c. Decare a map of CM services definition
+
+We expose a map of all the versioned supported. Initially we'll have a single version but as our content evolves we will be adding more versions in this map.
+
+```ts
+// common/content_management/cm_services.ts
+import type {
+  ContentManagementServicesDefinition as ServicesDefinition,
+  Version,
+} from '@kbn/object-versioning';
+
+// We export the versionned service definition from this file and not the barrel to avoid adding
+// the schemas in the "public" js bundle
+
+import { serviceDefinition as v1 } from './v1/cm_services';
+
+export const cmServicesDefinition: { [version: Version]: ServicesDefinition } = {
+  1: v1,
+};
+```
+
+### 3. Create a Storage instance for the content
+
+Once we have all our TS types defined and our CM ServicesDetinition map, we can create a `ContentStorage` class and its CRUD + search methods.
+
+```ts
+/**
+ * Import the map of CM services definitions that we created earlier.
+ */
+import { cmServicesDefinition } from '../../common/content_management/cm_services';
+
+/**
+ * It is a good practice to not directly exposes the SO document fields, specially the "attributes" object.
+ * Having a serializer function to convert the SavedObject<T> to our own specific content (MapItem) guarantees
+ * that we won't leak any additional fields in our Response, even when the SO client adds new fields to its responses.
+ */ 
+function savedObjectToMapItem(
+  savedObject: SavedObject<MapSavedObjectAttributes>
+): MapItem {
+  const {
+    id,
+    type,
+    updated_at: updatedAt,
+    created_at: createdAt,
+    attributes: { title, description, layerListJSON, mapStateJSON, uiStateJSON },
+    references,
+    error,
+    namespaces,
+  } = savedObject;
+
+  return {
+    id,
+    type,
+    updatedAt,
+    createdAt,
+    attributes: {
+      title,
+      description,
+      layerListJSON,
```

Comment:
Should this ideally be objects with a proper schema instead of serialized strings? 
I understand that this is likely very difficult for initial migration to CM, but maybe we should mention that this is the end goal